src/screencastPanel.ts

```diff
@@ -2,30 +2,41 @@
 // Licensed under the MIT License.
 import * as path from 'path';
 import * as vscode from 'vscode';
-import { encodeMessageForChannel, WebSocketEvent } from './common/webviewEvents';
+import {
+    encodeMessageForChannel,
+    WebSocketEvent,
+    ITelemetryProps,
+    ITelemetryMeasures,
+} from './common/webviewEvents';
 import { PanelSocket } from './panelSocket';
 import { ScreencastView } from './screencast/view';
 import {
     SETTINGS_STORE_NAME,
     SETTINGS_SCREENCAST_WEBVIEW_NAME,
 } from './utils';
+import TelemetryReporter from 'vscode-extension-telemetry';
 
 export class ScreencastPanel {
     private readonly context: vscode.ExtensionContext;
     private readonly extensionPath: string;
     private readonly panel: vscode.WebviewPanel;
+    private readonly telemetryReporter: TelemetryReporter;
```

Comment:
We should consider making this a singleton that's exported by the telemetry module instead of passing the ref around.

Reply:
I like this idea and was going to follow up to ask if you would want to break this out in a separate change, but I guess your approval answers that.

We would probably need to make a wrapper class for the TelemetryReporter since we cannot modify it directly.  We can also move some functions to this wrapper class and cut down on duplicate code.